docs/UserGuide.adoc

```diff
@@ -377,57 +412,83 @@ TODO: ADD DETAILS ONCE IMPLEMENTED
 [[FilterStudent]]
 ==== Filter `filter`
 
-TODO: ADD DETAILS ONCE IMPLEMENTED
+Filter Student returns you all the students according to the given keywords below.
+
+You can filter students in a class using these two ways:
+
+****
+1. module code and group code `m/` `g/`
+2. module code `m/`
+****
+
+1. Using both module code and group code returns you all the students specified by the group code and module code.
+
+Format: `student filter [m/MOD_CODE] g/GROUP_CODE`
+
+Examples:
+
+* ` student filter m/cs2103t g/g06`
+Filters all students in module `CS2103T`, under group `G06`.
+
+2. Using just the module code will return you all the students from the first group of module code shown in UI.
+
+Format: `student filter m/MOD_CODE`
+
+Examples:
+
+* `student filter m/cs2103t`
+Filters all the students in module `CS2103T`, first group of CS2103T that is currently showing in UI.
+
+[NOTE]
+Keywords are case-insensitive. eg. `cs2103t` will match `CS2103T`
+
 
 === Session View
 
 [[AddSession]]
 ==== Adding a Session `session add`
 
-TODO - UPDATE
+Adds a new session. The new undone session will be shown under Session view.
```

Comment:
Perhaps you could frame it like this:

The new session will be shown under Session View and will automatically be marked as un-done.